### PR TITLE
Milestone 3

### PR DESCRIPTION
**Pull request by RobertTamas**

### CODE REVIEW

File: api/src/main/java/cz/muni/fi/facade/ComponentFacade.java [hunk 1]
```diff
@@ -59,14 +59,14 @@
      *
      * @param component non-null object to be created in a database
      */
-    void createComponent(ComponentDTO component);
+    ComponentDTO createComponent(ComponentDTO component);
```
Comment (matusmacko) [inline api/src/main/java/cz/muni/fi/facade/ComponentFacade.java]: add `@return` to javadoc

File: api/src/main/java/cz/muni/fi/facade/ComponentFacade.java [hunk 2]
```diff
@@ -59,14 +59,14 @@
...
 
     /**
      * Takes an object of type Component and updates an entry in a database
      *
      * @param component non-null object to be updated in a database
      */
-    void updateComponent(ComponentDTO component);
+    ComponentDTO updateComponent(ComponentDTO component);
```
Comment (matusmacko) [inline api/src/main/java/cz/muni/fi/facade/ComponentFacade.java]: add `@return` to javadoc

File: service/src/main/java/cz/muni/fi/service/ComponentService.java [hunk 3]
```diff
@@ -70,7 +74,7 @@
      * @throws IllegalArgumentException when component argument is null
      * @throws cz.muni.fi.service.exception.ServiceDataAccessException if there is problem with dao
      */
-    void createComponent(Component component);
+    Component createComponent(Component component);
```
Comment (matusmacko) [inline service/src/main/java/cz/muni/fi/service/ComponentService.java]: update javadoc

File: service/src/main/java/cz/muni/fi/service/ComponentService.java [hunk 4]
```diff
@@ -79,7 +83,7 @@
      * @throws IllegalArgumentException when component argument is null
      * @throws cz.muni.fi.service.exception.ServiceDataAccessException if there is problem with dao
      */
-    void updateComponent(Component component);
+    Component updateComponent(Component component);
```
Comment (matusmacko) [inline service/src/main/java/cz/muni/fi/service/ComponentService.java]: update javadoc

File: mvc/src/main/java/cz/muni/fi/mvc/controllers/ComponentController.java [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+public class ComponentController {
+
+    private final static Logger log = LoggerFactory.getLogger(ComponentController.class);
+
+    @Autowired
+    ComponentFacade componentFacade;
+
+    @RequestMapping(value = "/create", method = RequestMethod.GET)
+    public String createComponent(Model model,  HttpServletRequest request, UriComponentsBuilder uriBuilder, RedirectAttributes redirectAttributes) {
+        //String res = Tools.redirectNonTestDriver(request, uriBuilder, redirectAttributes);
```
Comment (matusmacko) [inline mvc/src/main/java/cz/muni/fi/mvc/controllers/ComponentController.java]: I suppose we will use these commented parts in security part. Am I right?

File: mvc/src/main/java/cz/muni/fi/mvc/controllers/ComponentController.java [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+    }
+
+    @RequestMapping(value = "/create", method = RequestMethod.POST)
+    public String create(@Valid @ModelAttribute("componentCreate")ComponentDTO formBean, BindingResult bindingResult,
+                         Model model, RedirectAttributes redirectAttributes, UriComponentsBuilder uriBuilder, HttpServletRequest request) {
+
+        //String res = Tools.redirectNonTestDriver(request, uriBuilder, redirectAttributes);
+        //if (res != null) return res;
+
+        if (componentFacade.findComponentByName(formBean.getName()) != null) {
```
Comment (matusmacko) [inline mvc/src/main/java/cz/muni/fi/mvc/controllers/ComponentController.java]: store this in separate variable and don't make 2 same db calls..

File: mvc/src/main/java/cz/muni/fi/mvc/controllers/ComponentController.java [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+        return "components/read";
+    }
+
+    @RequestMapping(value="", method = RequestMethod.GET)
+    public String list(@Valid @ModelAttribute("filter")ComponentFilter filterBean, Model model, HttpServletRequest request, UriComponentsBuilder uriBuilder) {
+
+        ComponentFilter.ComponentFilterType filterNone = ComponentFilter.ComponentFilterType.NONE;
+
+        if (filterBean.getAvailable()) {
+            if (filterBean.getType() == filterNone || filterBean.getType() == null) {
```
Comment (matusmacko) [inline mvc/src/main/java/cz/muni/fi/mvc/controllers/ComponentController.java]: these blocks in if and else are almost the same. pls refactor without code repetition

**Comment by matusmacko** [comment]
@RobertTamas pls rebase

File: mvc/src/main/java/cz/muni/fi/mvc/controllers/ComponentController.java (hunk 5 above)
Comment (RobertTamas) [reply mvc/src/main/java/cz/muni/fi/mvc/controllers/ComponentController.java]: yep

**Pull request merged**